internal/pkg/skuba/addons/gangway_test.go

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package addons
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/SUSE/skuba/internal/pkg/skuba/kubernetes"
+	"github.com/SUSE/skuba/internal/pkg/skuba/skuba"
+	img "github.com/SUSE/skuba/pkg/skuba"
+	"k8s.io/apimachinery/pkg/util/version"
+)
+
+func TestGetGangwayImage(t *testing.T) {
+	tests := []struct {
+		name     string
+		imageTag string
+		want     string
+	}{
+		{
+			name:     "TestGetGangwayInitImage",
+			imageTag: "3.1.0",
+			want:     img.ImageRepository + "/gangway:3.1.0",
+		},
+		{
+			name:     "TestGetGangwayInitImage_rev",
+			imageTag: "3.1.0-rev2",
+			want:     img.ImageRepository + "/gangway:3.1.0-rev2",
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetGangwayImage(tt.imageTag); got != tt.want {
+				t.Errorf("GetGangwayImage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_renderContext_GangwayImage(t *testing.T) {
+	tests := []struct {
+		name          string
+		renderContext renderContext
+		want          string
+	}{
+		{
+			name: "Test_renderContext_GangwayImage_1.15.2",
+			renderContext: renderContext{
+				config: AddonConfiguration{
+					ClusterVersion: version.MustParseSemantic("1.15.2"),
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: img.ImageRepository + "/gangway:[[:digit:]]{1,}.[[:digit:]]{1,}.[[:digit:]]{1,}[-rev[:digit:]{1,}]{0,1}",
+		},
+		{
+			name: "Test_renderContext_GangwayImage_1.16.2",
+			renderContext: renderContext{
+				config: AddonConfiguration{
+					ClusterVersion: version.MustParseSemantic("1.16.2"),
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: img.ImageRepository + "/gangway:[[:digit:]]{1,}.[[:digit:]]{1,}.[[:digit:]]{1,}[-rev[:digit:]{1,}]{0,1}",
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.renderContext.GangwayImage()
+			matched, err := regexp.Match(tt.want, []byte(got))
+			if err != nil {
+				t.Errorf(err.Error())
```

Comment:
t.Error(err) or t.Errorf(“%v”, err)

Reply:
what is reasoning of using t.Error(err) or t.Errorf(“%v”, err)?

Reply:
Well, Errorf is `Errorf(format string, a ...interface{})`. Since this does not pass format, so no need suffix `f`